cypress-io/cypress
# feat(launchpad): add design tokens from Figma

- Closes https://cypress-io.atlassian.net/browse/UNIFY-360


### Additional details



### How this works

This adds the color hex codes from Figma to the colors available via Tailwind classes.

There are two ways to use the new colors:

#### 1. Directly use a color as a class

Example:

```html
<p class="text-jade-500"> Hello </p>
```

This works fine, and matches the exact way tailwind colors work. These colors do not have defaults the way that the functionally-named variants like "primary" do below.

#### 2. Use a color indirectly via a theme "role"

Example:

```html
<p class="text-primary"> Hello </p>
```

This maps to the color `indigo` and has a default defined of `indigo-500`. 

The following also works:

```html
<p class="bg-primary-50"> Hello </p>
```

This means we can use various shades of a named theme color without having to create named aliases like "primary-light-3" or something. We can choose to add aliases for common variations as needed, but we don't have to.

#### Intended use

The goal here is to be able to opt-in to meaningful color names when we judge that usage of a color is linked to that color's function in the design as the primary/secondary color or whatever. If there is a non-themed usage of a color, we can directly reference that color by name. This way we can as much as possible group colors that should change together, together.


### How to test

After checking out this branch, run `yarn` and then `yarn dev`. See that colors display correctly with a class like `text-primary`. The "browse manually" text is in the global mode landing page is a good example of this.

This PR also adds the `windicss-analyzer` to launchpad, so that can be tested by running `yarn windi` in the launchpad directory and visiting the generated site that it serves on localhost. That will create a report like this.

![windicss report](https://user-images.githubusercontent.com/8340719/134568606-d6ff5e20-a562-4979-8af9-e775ff9f572b.png)
